assayist/processor/base.py

```diff
@@ -136,17 +172,86 @@ def create_or_update_archive_artifact(self, archive_id, filename, arch, archive_
             _type = 'other'
         return self.__create_or_update_artifact(archive_id, _type, arch, filename, checksum)
 
-    def create_or_update_source_location(self, url, canonical_version):
+    def create_or_update_source_location(self, url, component, canonical_version=None):
         """
         Create or update SourceLocation.
 
+        Connect it to the component provided.
+        If canonical_version is provided link it with similar SourceLocations via SUPERSEDES.
+
         :param str url: the url and possibly commit hash of the source location
+        :param Component componet: the component associated with this SourceLocation
         :param str canonical_version: (optional) the version of the component that
                                       corresponds with this commit id
         :return: a SourceLocation object
         :rtype: assayist.common.models.source.SourceLocation
         """
-        sl = source.SourceLocation.create_or_update({'url': url})[0]
+        # Figure out what type of SourceLocation we're dealing with.
+        sl_type = ''
+        if canonical_version:
+            # TODO: A better determination?
+            if '.redhat.com/' in url:
+                sl_type = 'local'
+            else:
+                sl_type = 'upstream'
```

Comment:
Should the caller provide this?

Reply:
Only if we different analysers would want to use different criteria to determine if a source location is upstream or not. If so we can easily refactor it, but for now I'll leave it.